### ISSUE TITLE
toolchainopts openmp = True on cray adds "-True" to CFLAGS & Friends

### ISSUE THREAD

**Issue by gppezzi** [hpcugent/easybuild-framework#1700]
And the Cray wrapper doesn't like '-True' as flag

**Comment by boegel** [comment]
The problem is that `openmp` is not included as a key in `COMPILER_UNIQUE_OPTION_MAP` in `toolchains/compiler/craype.py`

We have a handful of them listed in the `CrayPECompiler` class, and we're injecting the ones for the precision flags for GCC/Intel compilers resp. when constructing a `CrayPE*` instance, but `openmp` is simply unknown.

This also highlights a bug in the toolchain mechanism somewhere, since it should clearly never be using `-True`, that's just stupid. Tracking down this one may be a bit harder though... @pforai?

@gppezzi What's the correct way of instructing the Cray compiler wrappers to enable OpenMP?

**Comment by boegel** [comment]
fixed in #1794, so good to close @gppezzi?

**Comment by gppezzi** [comment]
As for the openmp issue, yes you can close this (when #1794 is merged). 

But is that the expected behaviour for values that are not defined on the toolchain options map? To just pass the content of whatever we put there as argument to the *FLAGS* variables?

**Comment by boegel** [comment]
@gppezzi yeah, good point, we could do better here... I'll try and take a look to throw a more meaningful error.
